File: matrices_tf.js
function generarMatriz(filas, columnas) {
    return tf.randomUniform([filas, columnas], 1, 21, 'int32');
}

function operacionesConMatrices(filas, columnas) {
    const A = generarMatriz(filas, columnas);
    const B = generarMatriz(filas, columnas);

    const suma = tf.add(A, B);
    const multiplicacion = tf.matMul(A, B);

    Promise.all([
        A.array(),
        B.array(),
        suma.array(),
        multiplicacion.array()
    ]).then(([a, b, s, m]) => {
        console.log("Matriz A:", a);
        console.log("Matriz B:", b);
        console.log("Suma:", s);
        console.log("Multiplicación:", m);
    });
}

operacionesConMatrices(3, 3);

File: matrices_tf.py
import tensorflow as tf


def generar_matriz(dim):
    return tf.random.uniform(shape=dim, minval=1, maxval=21, dtype=tf.int32)


def operaciones_con_matrices(dim):
    A = generar_matriz(dim)
    B = generar_matriz(dim)

    suma = tf.add(A, B)
    multiplicacion = tf.matmul(A, B)

    return A, B, suma, multiplicacion


dimension = (3, 3)


A, B, suma, multiplicacion = operaciones_con_matrices(dimension)


print("Matriz A:", A.numpy())
print("Matriz B:", B.numpy())
print("Suma:", suma.numpy())
print("Multiplicación:", multiplicacion.numpy())